desktop/core/src/desktop/js/apps/admin/Overview/Analytics.tsx

```diff
@@ -14,63 +14,106 @@
-    } else {
-      const errorMessage = collectUsage
-        ? t('Failed to activate analytics.')
-        : t('Failed to deactivate analytics.');
-      huePubSub.publish('hue.global.error', { message: errorMessage });
+  useEffect(() => {
+    if (data?.analytics_enabled !== undefined) {
+      setAnalyticsEnabled(data.analytics_enabled);
     }
-  };
+  }, [data]);
+
+  const { save: saveAnalyticsPreference, loading: savingPreference } =
+    useSaveData<UsageAnalyticsResponse>(UPDATE_USAGE_ANALYTICS_API_URL, {
+      onSuccess: response => {
+        if (response.analytics_enabled !== undefined) {
+          reloadData();
+          setAnalyticsEnabled(response.analytics_enabled);
+          const successMessage = response.analytics_enabled
+            ? t('Analytics have been activated.')
+            : t('Analytics have been deactivated.');
+          huePubSub.publish('hue.global.info', { message: successMessage });
+        }
+      },
+      onError: error => {
+        console.error('Error updating usage analytics settings:', error);
+        const errorMessage = analyticsEnabled
+          ? t('Failed to deactivate analytics.')
+          : t('Failed to activate analytics.');
+        huePubSub.publish('hue.global.error', { message: errorMessage });
+      },
+      postOptions: {
+        headers: {
+          'Content-Type': 'application/json',
+        }
+      }
+    });
+  
+const handleCheckboxChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+  const newPreference = event.target.checked;
 
-  const handleCheckboxChange = event => {
-    const newPreference = event.target.checked;
-    setCollectUsage(newPreference);
-    saveCollectUsagePreference(newPreference);
+
+  const payload = {
+    analytics_enabled: newPreference 
   };
 
+  try {
+    await saveAnalyticsPreference(payload);
+    console.log('Payload sent to server:', payload);
+  } catch (err) {
+    console.error('Error during saveAnalyticsPreference:', err);
+  }
+};
+
   return (
-    <div className="overview-analytics">
-      <h3>{t('Anonymous usage analytics')}</h3>
-      <div className="analytics-checkbox-container">
-        <Input
-          type="checkbox"
-          className="analytics__checkbox-icon"
-          id="usage_analytics"
-          checked={collectUsage}
-          onChange={handleCheckboxChange}
-        />
-        <label htmlFor="usage_analytics" className="usage__analytics">
-          {t('Help improve Hue with anonymous usage analytics.')}
-        </label>
+    <LoadingErrorWrapper loading={loading || savingPreference}>
+      <div className="overview-analytics">
+        <h3>{t('Anonymous usage analytics')}</h3>
+
+        <div className="analytics-checkbox-container">
+          <Input
+            type="checkbox"
+            className="analytics__checkbox-icon"
+            id="usage_analytics"
+            checked={analyticsEnabled}
+            onChange={handleCheckboxChange}
```

Comment:
suggestion: you could simplify this by just calling saveAnalyticsPreference in an arrow function here
`(event)=> saveAnalyticsPreference({analytics_enabled: event.target.checked})`